RFCs/0031-Multi-Test-Run-Finalization.md

```diff
@@ -0,0 +1,304 @@
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the finalization is canceled or not.
+        /// </summary>
+        [DataMember]
+        public bool IsCanceled { get; private set; }
+
+        /// <summary>
+        /// Gets the error encountered during the finalization of the test runs. Null if there is no error.
+        /// </summary>
+        [DataMember]
+        public Exception Error { get; private set; }
+
+        /// <summary>
+        /// Get or Sets the Metrics
+        /// </summary>
+        [DataMember]
+        public IDictionary<string, object> Metrics { get; set; }
+    }
+}
+```
+
+2. `MultiTestRunFinalizationProgressEventArgs` used by IMultiTestRunFinalizationEventsHandler:
+
+```
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel.Client
+{
+    [DataContract]
+    public class MultiTestRunFinalizationProgressEventArgs : EventArgs
+    {
+        /// <summary>
+        /// Default constructor.
+        /// </summary>
+        /// <param name="currentProcessorIndex">Specifies current processor index.</param>
+        /// <param name="currentProcessorUris">Specifies current processor Uris.</param>
+        /// <param name="currentProcessorProgress">Specifies current processor progress.</param>
+        /// <param name="processorsCount">Specifies the overall number of processors.</param>
+        public MultiTestRunFinalizationProgressEventArgs(long currentProcessorIndex, ICollection<Uri> currentProcessorUris, long currentProcessorProgress, long processorsCount)
+        {
+            CurrentProcessorIndex = currentProcessorIndex;
+            CurrentProcessorUris = currentProcessorUris;
+            CurrentProcessorProgress = currentProcessorProgress;
+            ProcessorsCount = processorsCount;
+        }
+
+        /// <summary>
+        /// Gets a current processor index.
+        /// </summary>
+        [DataMember]
+        public long CurrentProcessorIndex { get; private set; }
+
+        /// <summary>
+        /// Gets a current processor URI.
+        /// </summary>
+        [DataMember]
+        public ICollection<Uri> CurrentProcessorUris { get; private set; }
+
+        /// <summary>
+        /// Gets a current processor progress.
+        /// </summary>
+        [DataMember]
+        public long CurrentProcessorProgress { get; private set; }
+
+        /// <summary>
+        /// Gets the overall number of processors.
+        /// </summary>
+        [DataMember]
+        public long ProcessorsCount { get; private set; }
+    }
+}
+```
+
+3. Class `MultiTestRunFinalizationPayload` defines message payload connected to message ID `"MultiTestRunFinalization.Start"`
+
+```
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel.Client
+{
+    /// <summary>
+    /// Class used to define the MultiTestRunFinalizationPayload sent by the Vstest.console translation layers into design mode
+    /// </summary>
+    public class MultiTestRunFinalizationPayload
```

Comment:
>MultiTestRunFinalizationPayload [](start = 17, length = 31)

We probably could defer this to the editor API specification doc I was referring to earlier. That can happen in a separate PR as well if you'd prefer so you wouldn't not to block any code changes based on that.

Reply:
Created new task for it.